### PR TITLE
Support for collectable cache + code refactors

### PR DESCRIPTION
**Pull request by dkraczkowski**

### CODE REVIEW

File: chocs_middleware/cache/middleware.py [hunk 1]
```diff
@@ -100,22 +112,31 @@ def handle(self, request: HttpRequest, next: MiddlewareHandler) -> HttpResponse:
 
         if "vary" not in response.headers:
             response.headers["vary"] = ",".join(cache_vary)
-        else:
```
Comment (grahami) [inline chocs_middleware/cache/middleware.py:103]: Could it be the case that "vary" is in response.headers but "etag" is not?
I wonder if we would still need to generate a new cache_id in this case?

**Review by grahami — COMMENTED** [review]
Looks really nice with the cache revalidation.

Just one question about if/when to re-generate. the cache_id based on the response.

File: chocs_middleware/cache/middleware.py (hunk 1 above)
Comment (dkraczkowski) [reply chocs_middleware/cache/middleware.py:103]: @grahami I think you are right. I will prepare a fix.

**Comment by dkraczkowski** [comment]
@grahami https://github.com/kodemore/chocs-cache/pull/1/files#diff-ab95759879184929a4372c87029f2ec36debf3937c376c11c2e706dcbc381106R387
Please have a look at the above test and let me know if you think this will do. Thanks ;)

**Review by szymonmiks-netguru — APPROVED** [review]

**Review by grahami — APPROVED** [review]
Looks great. Thanks for helping me understand

**Pull request merged**